File: src/app/Model/Owner.ts
import {Entity} from './Entity';
import {Animal} from './Animal';

export class Owner extends Entity{
  name: string;
  email: string;
  animals: Animal[] = [];
  constructor(id ?: number, name ?: string, email ?: string, animals ?: Animal[]) {
    super(id);
    this.name = name;
    this.email = email;
    this.animals = animals;
  }
}

File: monitor-core/src/main/java/monitorCore/Model/Owner.java
package monitorCore.Model;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.UniqueConstraint;
import java.io.Serializable;
import java.util.ArrayList;

@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Builder
@Entity(name="Owner")
public class Owner extends BaseEntity implements Serializable {
    private String name;
    private String email;
    ArrayList<Animal> animals;
}
